openid-4-verifiable-presentations-1_0.md

```diff
@@ -432,54 +427,26 @@ Note: "https://self-issued.me/v2" is a symbolic string and can be used as an `au
-
-This specification defines the following values for the `client_id_scheme` parameter, followed by the examples where applicable: 
+This specification defines additional mechanisms to enable Wallets to determine whether the provided `redirect_uri` and `response_uri` parameters and public verification keys belong to a particular Client, utilizing existing mechanisms such as Dynamic Client Registration [@!RFC7591], OpenID Federation [@!OpenID.Federation], and the `client_metadata` parameter defined by this specification. Additional Verifier Metadata is retrieved using existing mechanisms or the `client_metadata` parameter.
 
-* `pre-registered`: This value represents the [@!RFC6749] default behavior, i.e., the Client Identifier needs to be known to the Wallet in advance of the Authorization Request. The Verifier metadata is obtained using [@!RFC7591] or through out-of-band mechanisms.
+The Wallet MUST verify that the provided `redirect_uri` or `response_uri` belongs to the Client.
 
-* `redirect_uri`: This value indicates that the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used) is also the value of the Client Identifier. The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter (or `response_uri` when Response Mode `direct_post` is used). All Verifier metadata parameters MUST be passed using the `client_metadata` parameter defined in (#vp_token_request).
+If the Authorization Request is signed, the Wallet MUST authenticate the Request Object against the `client_id` contained in the Request Object. In this case, any `jwks` parameter in the `client_metadata` parameter MUST NOT be used to verify the signature of the Request Object.
 
-The following is a non-normative example of a request when `client_id` equals `redirect_uri`.
-
-```
-HTTP/1.1 302 Found
-Location: https://client.example.org/universal-link?
-  response_type=vp_token
-  &client_id=https%3A%2F%2Fclient.example.org%2Fcb
-  &client_id_scheme=redirect_uri
-  &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb
-  &presentation_definition=...
-  &nonce=n-0S6_WzA2Mj
-  &client_metadata=%7B%22vp_formats%22:%7B%22jwt_vp_json%22:%
-  7B%22alg%22:%5B%22EdDSA%22,%22ES256K%22%5D%7D,%22ldp
-  _vp%22:%7B%22proof_type%22:%5B%22Ed25519Signature201
-  8%22%5D%7D%7D%7D
-```
+If the Verifier Metadata is not obtained from a trusted source and if the Authorization Request is unsigned, the Wallet MUST verify that the `client_id` matches the `redirect_uri` or `response_uri` parameter. Examples for trusted sources include prior registration (e.g., using [@!RFC7591]) or other out-of-band processes that allow the Wallet to retrieve trusted Verifier Metadata for a particular Client.
```

Comment:
Is this method always enabled?

Reply:
No, it is not. We need adapt the language, also @martijnharing had a similar comment here https://github.com/openid/OpenID4VP/pull/237#discussion_r1741930279